core/src/main/java/io/questdb/std/CompactIntHashSet.java

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.std;
+
+import java.util.Arrays;
+
+/**
+ * Unlike {@link IntHashSet} doesn't keep an additional list for faster iteration and index-based access
+ * and also has a slightly higher load factor.
+ */
+public class CompactIntHashSet extends AbstractIntHashSet {
+
+    private static final int MIN_INITIAL_CAPACITY = 16;
```

Comment:
Should this be 1 or something lower than 16 to save the size?

Reply:
Current initial size is much more compact than the old `IntList`. It should be enough to fit 9 items. I'm not sure if it makes sense to fine-tune initial size and load factor without knowing anything about data. Why do you think we should shrink it? Java object headers themselves are quite something, so setting capacity to 1 would lead to an awful metadata:data ratio.